File: 수학(구현)/백준_4948_베르트랑 공준/베르트랑 공준.kt
import java.util.*

fun is_prime(n: Int): Boolean {
    for (i in 2..n) {
        if (i * i > n) break
        if (n % i == 0) return false
    }
    return true
}

fun main(args: Array<String>) {
    val s = Scanner(System.`in`)
    var n = 0
    while (true) {
        n = s.nextInt()
        if (n == 0) break
        var count = 0

        for (i in n + 1..n * 2) {
            if (is_prime(i)) count++
        }
        println(count)
    }
}

File: 수학(구현)/백준_4948_베르트랑 공준/베르트랑 공준.cs
using System.Collections;
using static System.Console;
using System;

class Program
{
    static bool is_prime(int n)
    {
        for (int i = 2; i * i <= n; i++)
        {
            if (n % i == 0) return false;
        }
        return true;
    }

    public static void Main()
    {
        int n;
        int s;
        while (true)
        {
            s = int.Parse(ReadLine());
            if (s == 0) break;

            int count = 0;
            for (int i = s + 1; i <= 2 * s; i++)
            {
                if (is_prime(i)) count++;
            }
            WriteLine(count);
        }


    }
}